core/controller/src/main/scala/whisk/core/loadBalancer/LoadBalancerService.scala

```diff
@@ -89,81 +87,43 @@ class LoadBalancerService(
+            val subject = msg.user.subject.asString
+            val entry = setupActivation(msg.activationId, subject, invokerName, timeout, transid)
+            logging.info(this, s"posting topic '$topic' with activation id '${msg.activationId}'")
+            producer.send(topic, msg).map { status =>
+                transid.finished(this, start, s"Posted to ${status.topic()}[${status.partition()}][${status.offset()}]")
+                entry.promise.future
+            }
         }
     }
 
-    /** Gets a producer which can publish messages to the kafka bus. */
-    private val producer = new KafkaProducerConnector(config.kafkaHost, executionContext)
-
-    private val invokerHealth = new InvokerHealth(config, invokerChangeCallback, () => producer.sentCount())
-
     /**
      * A map storing current activations based on ActivationId.
      * The promise value represents the obligation of writing the answer back.
      */
-    case class ActivationEntry(id: ActivationId, subject: String, invokerIndex: Int, created: Instant, promise: Promise[WhiskActivation])
+    case class ActivationEntry(id: ActivationId, subject: String, invokerName: String, created: Instant, promise: Promise[WhiskActivation])
     type TrieSet[T] = TrieMap[T, Unit]
     private val activationById = new TrieMap[ActivationId, ActivationEntry]
-    private val activationByInvoker = new TrieMap[Int, TrieSet[ActivationEntry]]
+    private val activationByInvoker = new TrieMap[String, TrieSet[ActivationEntry]]
     private val activationBySubject = new TrieMap[String, TrieSet[ActivationEntry]]
 
-    private val kv = new ConsulClient(config.consulServer)
-    private val reporter = new ConsulKVReporter(kv, 3 seconds, 10 seconds,
-        LoadBalancerKeys.hostnameKey,
-        LoadBalancerKeys.startKey,
-        LoadBalancerKeys.statusKey,
-        { _ =>
-            val activeCounts = getActiveCountByInvoker()
-            val totalActiveCount = activeCounts.values.sum
-            val health = invokerHealth.getInvokerHealth()
-            implicit val sid = TransactionId.loadbalancer
-            logging.info(this, s"In flight: $totalActiveCount = [${activeCounts.mkString(", ")}]")
-            logging.info(this, s"Invoker health: [${health.mkString(", ")}]")
-            Map(LoadBalancerKeys.invokerHealth -> getInvokerHealth(),
-                LoadBalancerKeys.activationCountKey -> producer.sentCount().toJson)
-        })
-
     private val consumer = new KafkaConsumerConnector(config.kafkaHost, "completions", "completed")
-    consumer.onMessage((topic, partition, offset, bytes) => {
+    consumer.onMessage((topic, _, _, bytes) => {
         val raw = new String(bytes, "utf-8")
-        CompletionMessage(raw) match {
-            case Success(m) => processCompletion(m)
-            case Failure(t) => logging.error(this, s"failed processing message: $raw with $t")
+        Messages.parse(raw) match {
+            case Success(m: CompletionMessage) => processCompletion(m)
```

Comment:
with all the pings from the invokers going through the same topic, we're going to inevitably delay active acks - one reason to perhaps have a separate topic for health/administrative functions.